test/SeedCppCoreTestUtilities/Mocks/Services/Model/MockAllergyModelService.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "SeedCppCore/Services/Model/IAllergyModelService.h"
+
+#include "SeedCppCore/Model/Allergy.h"
+#include "SeedCppCore/Model/AllergyMgr.h"
+
+
+namespace seed_cpp { namespace test_utility {
+
+	class MockAllergyModelService : public service::IAllergyModelService
+	{
+	public:
+		MockAllergyModelService();
+		virtual ~MockAllergyModelService();
+
+		MOCK_CONST_METHOD0(getEntityMgr, model::AllergyMgr&());
+		MOCK_CONST_METHOD2(getEntityById, const model::Allergy*(const std::string&, const model::LockableEntityMgrSubject::IReadLock&));
+
+		MOCK_METHOD2(addEntityProxy, const model::Allergy&(model::Allergy*, const model::LockableEntityMgrSubject::IWriteLock&));
```

Comment:
Not to change now, but as reference for the future. I think we should take advantadge of latest additions to gtest/gmock to avoid this proxy thing. Starting on version 1.8.1 if I'm not mistaken, this is no longer necessary.

Check:
* https://github.com/google/googletest/blob/master/googlemock/docs/cook_book.md
* Section: _Mocking Methods That Use Move-Only Types_

Reply:
Absolutely. As said, I won't change it now but I take note for further implementations.